File: bench/micro/form_vec_push.py
"""Python equivalent of form_vec_push.ap.
list.append from empty.
"""

import time

iters = 500_000
v = []
t0 = time.monotonic_ns()
for i in range(iters):
    v.append(i)
elapsed = time.monotonic_ns() - t0
print(f"iters={iters}")
print(f"len={len(v)}")
print(f"elapsed_ns={elapsed}")

File: bench/micro/form_vec_push.go
// Go equivalent of form_vec_push.ap.
// `append` to a `[]int` from cap=0; Go's growth policy is
// similar to Aperio's @form(vec) doubling.
package main

import (
	"fmt"
	"time"
)

func main() {
	iters := 500000
	v := make([]int, 0)
	t0 := time.Now()
	for i := 0; i < iters; i++ {
		v = append(v, i)
	}
	elapsed := time.Since(t0).Nanoseconds()
	fmt.Printf("iters=%d\n", iters)
	fmt.Printf("len=%d\n", len(v))
	fmt.Printf("elapsed_ns=%d\n", elapsed)
}
